internal/services/synapse/synapse_sql_pool_security_alert_policy_resource.go

```diff
@@ -0,0 +1,270 @@
+package synapse
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/synapse/mgmt/2021-03-01/synapse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSynapseSqlPoolSecurityAlertPolicy() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSynapseSqlPoolSecurityAlertPolicyCreateUpdate,
+		Read:   resourceSynapseSqlPoolSecurityAlertPolicyRead,
+		Update: resourceSynapseSqlPoolSecurityAlertPolicyCreateUpdate,
+		Delete: resourceSynapseSqlPoolSecurityAlertPolicyDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.SqlPoolSecurityAlertPolicyID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"sql_pool_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.SqlPoolID,
+			},
+
+			"disabled_alerts": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Set:      pluginsdk.HashString,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"Sql_Injection",
+						"Sql_Injection_Vulnerability",
+						"Access_Anomaly",
+						"Data_Exfiltration",
+						"Unsafe_Action",
+					}, false),
+				},
+			},
+
+			"email_account_admins": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"email_addresses": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
+				Set: pluginsdk.HashString,
+			},
+
+			"retention_days": {
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				Default:      0,
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+
+			"state": {
```

Comment:
Could we be explicit here and rename this to 
```suggestion
			"policy_state": {
```